src/sentry/api/helpers/group_index.py

```diff
@@ -751,6 +752,53 @@ def update_groups(request, group_ids, projects, organization_id, search_fn):
                                         "status": GroupResolution.Status.resolved,
                                     }
                                 )
+                            else:
+                                # If we already know the `next` release in date based ordering
+                                # when clicking on `resolvedInNextRelease` because it is already
+                                # been released, there is no point in setting GroupResolution to
+                                # be of type in_next_release but rather in_release would suffice
+
+                                try:
+                                    # Get current release object from current_release_version
+                                    current_release_obj = Release.objects.get(
+                                        version=current_release_version,
+                                        organization_id=projects[0].organization_id,
+                                    )
+
+                                    date_order_q = Q(
+                                        date_added__gt=current_release_obj.date_added
+                                    ) | Q(
+                                        date_added=current_release_obj.date_added,
+                                        id__gt=current_release_obj.id,
+                                    )
+
+                                    # Find the next release after the current_release_version
+                                    # i.e. the release that resolves the issue
+                                    resolved_in_release = (
+                                        Release.objects.filter(
+                                            date_order_q,
+                                            projects=projects[0],
+                                            organization_id=projects[0].organization_id,
+                                        )
+                                        .order_by("date_added", "id")[:1]
+                                        .get()
+                                    )
```

Comment:
Should we be using our usual coalesce of `date_released, date_added` here?

Reply:
I thought we stopped using that in favor of only date_added ? I remember we changed it in Organisation releases first

Reply:
Hmm, if we did so then there are a lot of places still using the old model. I don't really remember that decision?

Reply:
If I recall correctly, we opted for this solution because Coalescing `date_released` with `date_added` was giving us ambiguous ordering behavior where the `date_released` was taking precedence in the actual ordering while we show the `date_added` on the release card -> https://github.com/getsentry/sentry/pull/26256
And we have decided that moving forward we will do that